src/main/java/org/opensearch/geospatial/plugin/GeospatialPlugin.java

```diff
@@ -96,19 +105,26 @@ public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings sett
 
     @Override
     public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
+        this.datasourceFacade = new DatasourceFacade(parameters.client, parameters.ingestService.getClusterService());
+        this.geoIpDataFacade = new GeoIpDataFacade(parameters.ingestService.getClusterService(), parameters.client);
+        this.ip2GeoCache = new Ip2GeoCache(datasourceFacade);
```

Comment:
is it possible to inject those three service refs instead? this will make testing easier as we can inject mocks. In addition to that it's easy to adopt this code for later changes in case we'll have multiple implementation(s) for any of the services

Reply:
It is not possible to use dependency injection in processor factory creation. There is one option to avoid this parameter passing by using transport action inside processor. However, it will increase latency and code complexity.